scg-system/src/main/java/io/telicent/core/SmartCacheGraph.java

```diff
@@ -52,6 +64,40 @@ public static FusekiServer.Builder smartCacheGraphBuilder() {
      * Builder for a Fuseki server configured for SmartCacheGraph
      */
     public static FusekiServer construct(String... args) {
+        YAMLConfigParser ycp = new YAMLConfigParser();
+        RDFConfigGenerator rcg = new RDFConfigGenerator();
+        String configPath;
+        String pattern = ".*\\.(yaml|yml)$";
+        Pattern regex = Pattern.compile(pattern);
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].equalsIgnoreCase("--conf") && i + 1 < args.length ) {
+                configPath = args[i + 1];
+                Matcher matcher = regex.matcher(configPath);
+                if (matcher.matches()) {
+                    try {
+                        ConfigStruct configStruct = ycp.runYAMLParser(configPath);
+                        Model configModel = rcg.createRDFModel(configStruct);
+
+                        File rdfConfigPath = File.createTempFile("RDF.state", ".ttl");
+                        rdfConfigPath.deleteOnExit();
+
+                        try (FileOutputStream out = new FileOutputStream(rdfConfigPath)) {
+                            configModel.write(out, "TTL", new File(configPath).getAbsoluteFile().getParentFile().toURI().toString());
+                            log.info(new File(configPath).getAbsoluteFile().getParentFile().toURI().toString());
+                            configModel.write(System.out, "TTL", new File(configPath).getAbsoluteFile().getParentFile().toURI().toString());
+                            args[i + 1] = rdfConfigPath.getPath();
+                        } catch (IOException e) {
+                            log.error(e.getMessage());
+                            throw new RuntimeException(e.getMessage());
+                        }
```

Comment:
I would be tempted to move this entire `try` block into a separate method. Just because there is a lot going on in this method already and it will make it easier to read and reason about. Something like `convertToRdf(yamlConfigPath)`.